plugins/outputs/cloudwatch/cloudwatch.go

```diff
@@ -151,67 +281,58 @@ func PartitionDatums(size int, datums []*cloudwatch.MetricDatum) [][]*cloudwatch
 	return partitions
 }
 
-// Make a MetricDatum for each field in a Point. Only fields with values that can be
-// converted to float64 are supported. Non-supported fields are skipped.
-func BuildMetricDatum(point telegraf.Metric) []*cloudwatch.MetricDatum {
-	datums := make([]*cloudwatch.MetricDatum, len(point.Fields()))
-	i := 0
+// Make a MetricDatum from telegraf.Metric. It would check if all required fields of
+// cloudwatch.StatisticSet are available. If so, it would build MetricDatum from statistic values.
+// Otherwise, fields would still been built independently.
+func BuildMetricDatum(buildStatistic bool, point telegraf.Metric) []*cloudwatch.MetricDatum {
```

Comment:
In general, changing an exported function's signature is advised against. Since nothing outside of this package uses it, maybe add a note that `BuildMetricDatum(telegraf.Metric)` is deprecated and create a new (unexported) `buildMetricDatum(bool, Metric)` function.

Reply:
This isn't something we are concerned about right now, maybe someday but not now.  The only API we provide stability for is the TOML file.  I keep meaning to document this, but haven't gotten to it.

Reply:
If this is not an issue, I could help to `unexport` all exported functions to avoid confusion.
How do you think?